src/ontodia/data/sparql/sparqlDataProvider.ts

```diff
@@ -0,0 +1,341 @@
+import 'whatwg-fetch';
+import { DataProvider, FilterParams } from '../provider';
+import { Dictionary, ClassModel, LinkType, ElementModel, LinkModel, LinkCount, PropertyModel } from '../model';
+import {
+    getClassTree,
+    getClassInfo,
+    getPropertyInfo,
+    getLinkTypes,
+    getElementsInfo,
+    getLinksInfo,
+    getLinksTypesOf,
+    getFilteredData,
+    getEnrichedElementsInfo,
+    getLinkTypesInfo,
+} from './responseHandler';
+import {
+    ClassBinding, ElementBinding, LinkBinding, PropertyBinding,
+    LinkTypeBinding, LinkTypeInfoBinding, ElementImageBinding, SparqlResponse, Triple,
+} from './sparqlModels';
+import {SparqlDataProviderSettings, OWLStatsSettings} from "./sparqlDataProviderSettings";
+
+export enum SparqlQueryMethod { GET = 1, POST }
+
+// this is runtime settings.
+export interface SparqlDataProviderOptions {
+
+    // sparql endpoint URL to use
+    endpointUrl: string;
+
+    // there are two options for fetching images: specify imagePropertyUris
+    // to use as image properties or specify a function to fetch image URLs
+
+    // properties to use as image URLs
+    imagePropertyUris?: string[];
+
+    // you can specify prepareImages function to extract image URL from element model
+    prepareImages?: (elementInfo: Dictionary<ElementModel>) => Promise<Dictionary<string>>;
+
+    // wether to use GET (more compatible, more error-prone due to large request URLs)
+    // or POST(less compatible, better on large data sets)
+    queryMethod?: SparqlQueryMethod;
+
+    // what property to use as instance labels. This will override dataLabelProperty from settings
+    labelProperty?: string;
+}
+
+export class SparqlDataProvider implements DataProvider {
+    dataLabelProperty: string;
+    constructor(private options: SparqlDataProviderOptions, private settings: SparqlDataProviderSettings = OWLStatsSettings) {
```

Comment:
I think it's better to make additional `sparqlSettings` option of type `SparqlDataProviderSettings` in `SparqlDataProviderOptions`.